File: src/ru/lionsoft/hello/scala/Point.scala
package ru.lionsoft.hello.scala

/**
 * Класс описывающий точку в 2D координатах
 * @param x абцисса
 * @param y ордината
 */
class Point(private var _x: Int = 0, private var _y: Int = 0) {

  private val bound = 100

  // Getters & Setters

  // Getter
  def x = _x
  // Setter
  def x_= (newValue: Int) = {
    if (newValue >= -bound && newValue <= bound) _x = newValue else printWarning
  }

  def y = _y
  def y_= (newValue: Int) = {
    if (newValue >= -bound && newValue <= bound) _y = newValue else printWarning
  }

  private def printWarning = println("WARNING: Out of bounds")

  /**
   * Переместить точку
   * @param dx смещение по абцисе
   * @param dy смещение по ординате
   */
  def move(dx: Int, dy: Int): Unit = {
    x += dx
    y += dy
  }

  // Cast to String
  override def toString: String = s"Point($x, $y)"

}

File: src/ru/lionsoft/hello/java/Point.java
package ru.lionsoft.hello.java;

public class Point {
    // Constant
    private static final int BOUND = 100;

    // Fields
    private int x;
    private int y;

    // Constructors
    public Point() {
    }

    public Point(int x) {
        this.x = x;
    }

//    public Point(int y) {
//        this.y = y;
//    }

    public Point(int x, int y) {
        this.x = x;
        this.y = y;
    }

    // Getters & Setters

    public int getX() {
        return x;
    }

    public void setX(int x) {
        if (x < -BOUND || x > BOUND) printWarning();
        this.x = x;
    }

    public int getY() {
        return y;
    }

    public void setY(int y) {
        if (y < -BOUND || y > BOUND) printWarning();
        this.y = y;
    }

    // Methods
    public void move(int dx, int dy) {
        x += dx;
        y += dy;
    }

    private void printWarning() {
        System.out.println("WARNING: Out of bounds");
    }

    @Override
    public String toString() {
        return "Point(" + x + ", " + y + ")";
    }
}
